domain/model/service/service.go

```diff
@@ -382,7 +382,7 @@ func (s *Service) DeleteModel(
 
 	// Delete common items from the model. This helps to ensure that the
 	// model is cleaned up correctly.
-	if err := s.st.Delete(ctx, uuid); err != nil {
+	if err := s.st.Delete(ctx, uuid); err != nil && !errors.Is(err, modelerrors.NotFound) {
```

Comment:
This is so we always attempts the DB deletion below, even if we don't think the model is there.